sys/linux/socket_netlink_route.txt

```diff
@@ -850,6 +866,37 @@ br_mdb_entry_sub_u [
 	in6_addr	ipv6_addr
 ]
 
+rtm_nh_policy [
+	NHA_ID		nlattr[NHA_ID, nha_id]
+	NHA_GROUP	nlattr[NHA_GROUP, array[nexthop_grp]]
+	NHA_GROUP_TYPE	nlattr[NHA_GROUP_TYPE, flags[nha_group_types, int16]]
+	NHA_BLACKHOLE	nlattr[NHA_BLACKHOLE, void]
+	NHA_OIF		nlattr[NHA_OIF, ifindex]
+	NHA_GATEWAY	nlattr[NHA_GATEWAY, nha_gateway]
+	NHA_ENCAP_TYPE	nlattr[NHA_ENCAP_TYPE, flags[lwtunnel_encap_types, int16]]
+	NHA_ENCAP	nlnest[NHA_ENCAP, lwtunnel_policy]
+	NHA_GROUPS	nlattr[NHA_GROUPS, void]
+	NHA_MASTER	nlattr[NHA_MASTER, nha_id]
+	NHA_FDB		nlattr[NHA_FDB, void]
+] [varlen]
+
+nexthop_grp {
+	id	nha_id
+	weight	int8
+	resvd1	const[0, int8]
+	resvd2	const[0, int16]
+}
+
+nha_gateway [
+	ip4		ipv4_addr
+	in6_addr	ipv6_addr
+]
+
+nha_id [
+	unassigned	const[0, int32]
+	id		proc[1, 2, int32]
```

Comment:
It's fine asking author during review if something is not obvious ;)
The chances are it's not obvious for other readers as well, maybe it deserves a comment then.

The reason for using proc in any context is if these IDs are stored in/checked against some global list so that one test process can collide with another test process.
Looking at this code:
https://elixir.bootlin.com/linux/v5.8-rc4/source/net/ipv4/nexthop.c#L1389
It seems these are allocated on a per-net-namespace basis and we always have separate net namespaces per test process. So unless there are other uses, there should be no collisions and then we could do simpler `type nha_id int32[0:2]`.
Albert, are there any global uses that can result in collisions? Or what was the motivation for proc?

Reply:
Was overthinking it, namespace separation suffices, changed to int32[0:2].